pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/loader/SegmentPreProcessor.java

```diff
@@ -130,7 +130,24 @@ public void process(@Nullable SegmentOperationsThrottler segmentOperationsThrott
         if (type != StandardIndexes.forward()) {
           IndexHandler handler = createHandler(type);
           indexHandlers.add(handler);
-          handler.updateIndices(segmentWriter);
+          try {
+            handler.updateIndices(segmentWriter);
+          } catch (IllegalStateException | UnsupportedOperationException e) {
+            // The index creation failure is expected in some cases, like when the index is not supported for the
+            // column or segment type.
+            // E.g. non-dictionary encoded TIMESTAMP type doesn't support range index, meanwhile dictionary encoded
```

Comment:
For this example, if we skip updating index, I believe query can fail during query time, and it will be extremely hard to reason about.

Reply:
No, the query won't fail.
The issue is that old segments wont be able to create index and the old segments won't be loaded.
Meanwhile the table config was updated long time back and very likely the user who restart the cluster (SRE) are not aware this change from the real user side(table maintainer).

Reply:
I agree we should provide a way to get out of this limbo state (best effort), but that should be under a flag similar to ingestion `continueOnError`. Also, we need to have a metric tracking index creation failures so that user is notified so that they can fix the table config accordingly.

Ideally such issues should be caught during table config validation. We should provide this flag just for temporary mitigation